src/pybind/mgr/dashboard/tests/test_ceph_service.py

```diff
@@ -85,14 +85,16 @@ def _get_smart_data(d):
     "by,args,log",
     [
         ('host', ('osd0',), 'from host osd0'),
-        ('daemon', ('osd', '1'), 'with ID 1')
+        ('daemon', ('osd', '1'), 'with ID 1'),
+        ('daemon', ('mon', '2'), 'with ID 2')
     ]
 )
 def test_get_smart_data(caplog, by, args, log):
     # pylint: disable=protected-access
     expected_data = {
         'aaa': {'device': {'name': '/dev/sda'}},
         'bbb': {'device': {'name': '/dev/sdb'}},
+        'ccc': {'device': {'name': '/dev/sdc'}},
```

Comment:
This is not testing your fix as the fixed method is already patched.
A test is needed to check that the appropriate ceph command is called depending on device type.
Something like this:
```
diff --git a/src/pybind/mgr/dashboard/tests/test_ceph_service.py b/src/pybind/mgr/dashboard/tests/test_ceph_service.py
index 6ff2199ade9..5b5fe5e1cf5 100644
--- a/src/pybind/mgr/dashboard/tests/test_ceph_service.py
+++ b/src/pybind/mgr/dashboard/tests/test_ceph_service.py
@@ -109,3 +109,22 @@ def test_get_smart_data(caplog, by, args, log):
             CephService._get_smart_data_by_device.assert_not_called()
             assert smart_data == {}
             assert log in caplog.text
+
+
+@mock.patch.object(CephService, 'send_command')
+def test_get_smart_data_from_appropriate_ceph_command(send_command):
+    send_command.side_effect = [
+        {'nodes': [{'name': 'osd.1', 'status': 'up'},{'name': 'mon.1', 'status': 'down'}]},
+        {'fake': {'device': {'name': '/dev/sda'}}}
+    ]
+    CephService._get_smart_data_by_device({'devid': '1', 'daemons': ['osd.1', 'mon.1']})
+    send_command.assert_has_calls([mock.call('mon', 'osd tree'),
+                                   mock.call('osd', 'smart', '1', devid='1')])
+
+    send_command.side_effect = [
+        {'nodes': [{'name': 'osd.1', 'status': 'down'}, {'name': 'mon.1', 'status': 'up'}]},
+        {'fake': {'device': {'name': '/dev/sda'}}}
+    ]
+    CephService._get_smart_data_by_device({'devid': '1', 'daemons': ['mon.1', 'osd.1']})
+    send_command.assert_has_calls([mock.call('mon', 'osd tree'),
+                                   mock.call('mon', 'device get-health-metrics', '1', devid='1')])
```

Reply:
Thanks a lot @alfonsomthd . Added this patch to the tests.